From 3a0544b42909fbd240cd508e7e452cb0903e73be Mon Sep 17 00:00:00 2001
From: Han Wang <416810799@qq.com>
Date: Sun, 23 Dec 2018 18:45:43 +0100
Subject: [PATCH 5/5] lineage-sdk: Fix onUpgrade() logic

 * The upgradeVersion < newVersion part is never reachable because we
   increase upgradeVersion unconditionally. Make it back to stage by
   only increase it when upgrading success.

Change-Id: Icac1e18c192292ad3147fd4af4c862e93ef34ba5
---
 .../lineagesettings/LineageDatabaseHelper.java         | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/packages/LineageSettingsProvider/src/org/lineageos/lineagesettings/LineageDatabaseHelper.java b/packages/LineageSettingsProvider/src/org/lineageos/lineagesettings/LineageDatabaseHelper.java
index fc8e2ce..1d9b525 100644
--- a/packages/LineageSettingsProvider/src/org/lineageos/lineagesettings/LineageDatabaseHelper.java
+++ b/packages/LineageSettingsProvider/src/org/lineageos/lineagesettings/LineageDatabaseHelper.java
@@ -224,11 +224,11 @@ public class LineageDatabaseHelper extends SQLiteOpenHelper{
                 loadStringSetting(stmt, LineageSettings.Secure.PROTECTED_COMPONENT_MANAGERS,
                         R.string.def_protected_component_managers);
                 db.setTransactionSuccessful();
+                upgradeVersion = 3;
             } finally {
                 if (stmt != null) stmt.close();
                 db.endTransaction();
             }
-            upgradeVersion = 3;
         }
 
         if (upgradeVersion < 4) {
@@ -248,12 +248,12 @@ public class LineageDatabaseHelper extends SQLiteOpenHelper{
                     loadIntegerSetting(stmt, LineageSettings.Global.WEATHER_TEMPERATURE_UNIT,
                             R.integer.def_temperature_unit);
                     db.setTransactionSuccessful();
+                    upgradeVersion = 5;
                 } finally {
                     if (stmt != null) stmt.close();
                     db.endTransaction();
                 }
             }
-            upgradeVersion = 5;
         }
 
         if (upgradeVersion < 6) {
@@ -283,6 +283,7 @@ public class LineageDatabaseHelper extends SQLiteOpenHelper{
                         stmt.execute();
                     }
                     db.setTransactionSuccessful();
+                    upgradeVersion = 7;
                 } catch (SQLiteDoneException ex) {
                     // LineageSettings.System.STATUS_BAR_CLOCK is not set
                 } finally {
@@ -290,7 +291,6 @@ public class LineageDatabaseHelper extends SQLiteOpenHelper{
                     db.endTransaction();
                 }
             }
-            upgradeVersion = 7;
         }
 
         if (upgradeVersion < 8) {
@@ -303,11 +303,11 @@ public class LineageDatabaseHelper extends SQLiteOpenHelper{
                 stmt.bindString(2, LineageSettings.Secure.PROTECTED_COMPONENT_MANAGERS);
                 stmt.execute();
                 db.setTransactionSuccessful();
+                upgradeVersion = 8;
             } finally {
                 if (stmt != null) stmt.close();
                 db.endTransaction();
             }
-            upgradeVersion = 8;
         }
 
         if (upgradeVersion < 9) {
@@ -357,12 +357,12 @@ public class LineageDatabaseHelper extends SQLiteOpenHelper{
                     stmt.bindString(1, LineageSettings.Secure.LINEAGE_SETUP_WIZARD_COMPLETED);
                     stmt.execute();
                     db.setTransactionSuccessful();
+                    upgradeVersion = 10;
                 } finally {
                     if (stmt != null) stmt.close();
                     db.endTransaction();
                 }
             }
-            upgradeVersion = 10;
         }
 
         if (upgradeVersion < 11) {
-- 
2.17.1

